filebeat/channel/connector.go

```diff
@@ -51,22 +54,28 @@ func (c *pipelineConnector) ConnectWith(cfg *common.Config, clientCfg beat.Clien
 		return nil, err
 	}
 
-	var err error
-	var userProcessors beat.ProcessorList
+	procs := processors.NewList(nil)
```

Comment:
to improve testability we should factor out the construction of the processor list into a separate function. Then we can can write tests with different configs and events only, without having to mock all pipeline specific functionality.